GitUI/GitUIExtensions.cs

```diff
@@ -145,15 +145,15 @@ void OpenWithDiffTool()
                         : $"Failed to get status for submodule \"{file.Name}\"";
                 }
 
-                var patch = await GetItemPatchAsync(fileViewer.Module, file, firstId, selectedId,
+                (Patch? patch, string? errorMessage)? result = await GetItemPatchAsync(fileViewer.Module, file, firstId, selectedId,
                     fileViewer.GetExtraDiffArguments(), fileViewer.Encoding);
 
                 cancellationToken.ThrowIfCancellationRequested();
                 return file.IsSubmodule
-                    ? LocalizationHelpers.ProcessSubmodulePatch(fileViewer.Module, file.Name, patch)
-                    : patch?.Text;
+                    ? LocalizationHelpers.ProcessSubmodulePatch(fileViewer.Module, file.Name, result.Value.patch)
+                    : result.Value.patch?.Text ?? result.Value.errorMessage;
```

Comment:
```suggestion
                    ? LocalizationHelpers.ProcessSubmodulePatch(fileViewer.Module, file.Name, patch)
                    : patch?.Text ?? errorMessage;
```
For submodule, the error is swallowed.

Reply:
The error message is set as the name, so it is somehow presented. This is a hack, should not occur very often especially for submodules (I considered ignoring them completely)